src/model/events.js

```diff
@@ -0,0 +1,56 @@
+import Observer from "../utils/observer.js";
+
+export default class Events extends Observer {
+  constructor() {
+    super();
+    this._events = [];
+  }
+
+  setEvents(events) {
+    this._events = events.slice();
+  }
+
+  getEvents() {
+    return this._events;
+  }
+
+  updateEvent(updateType, update) {
+    const index = this._events.findIndex((event) => event.id === update.id);
+
+    if (index === -1) {
+      throw new Error(`Can't update unexisting task`);
+    }
+
+    this._events = [
+      ...this._events.slice(0, index),
+      update,
+      ...this._events.slice(index + 1)
+    ];
+
+    this._notify(updateType, update);
+  }
+
+  addEvent(updateType, update) {
+    this._events = [
+      update,
+      ...this._events
+    ];
+
+    this._notify(updateType, update);
+  }
+
+  deleteEvent(updateType, update) {
+    const index = this._events.findIndex((event) => event.id === update.id);
+
+    if (index === -1) {
+      throw new Error(`Can't delete unexisting task`);
+    }
+
+    this._events = [
+      ...this._events.slice(0, index),
+      ...this._events.slice(index + 1)
+    ];
```

Comment:
Думаю, здесь лучше использовать `filter`:
```suggestion
    this._events = this._events.filter((_, i) => i !== index);
```

Reply:
+

Reply:
done